### PR TITLE
Support PSE emails.

### PR DESCRIPTION
**Pull request by AndresMorelos**
Support Scraping for PSE emails.

## PSE setting

```json
{
    "filters": [
        {
            "phrase": "Empresa:",
            "type": "EXPENSE"
        }
    ],
    "name": "PSE",
    "subject": "Confirmación Transacción PSE",
    "folder": "Cosas Personales/PSE",
    "ignore_phrase": "Estado de la transacción: Rechazada",
    "index_value": {
        "$numberLong": "0"
    }
}
```

### CODE REVIEW

File: AutomationServices/AutoMailChecker/utils.js [hunk 1]
```diff
@@ -1,34 +1,67 @@
...
-        if(value.indexOf(skipped_phrase) < 0){
+    if (value.indexOf(filter) > 0) {
+        if (value.indexOf(skipped_phrase) < 0) {
+            let description, textWithValue;
             const first = value.indexOf(filter)
-            const end = value.indexOf('*', first + 1)
-            const newValue = value.substring(first, end)
-            return newValue
+
+            if (value.indexOf('*', first + 1) > 0) {
```
Comment (FlavioAandres) [inline AutomationServices/AutoMailChecker/utils.js:46]: ~are you detecting the bancolombia bank message using the **56 for the account?? what happen in the case where appears the market name? 🤔🤔🤔~

Nevermind, the previous comment was before seeing that i wrote that piece of code :) 

What do you think about detect through the bank.name passed by params?

File: AutomationServices/AutoMailChecker/utils.js [hunk 2]
```diff
@@ -1,34 +1,67 @@
...
+                //BANCOLOMBIA
+                const end = value.indexOf('*', first + 1)
+                description = value.substring(first, end)
+                textWithValue = description
+            } else {
+                // PSE
+                const end = value.indexOf('[http://www.jlnsoftware.com.br', first + 1)
+                description = value.substring(first, end)
+                if (value.toLowerCase().includes('Valor de la Transacción:'.toLocaleLowerCase())){
```
Comment (FlavioAandres) [inline AutomationServices/AutoMailChecker/utils.js:55]: hey could you explain this? i mean, how works the PSE detection :)

File: serverless.yml [hunk 3]
```diff
@@ -8,30 +8,30 @@ provider:
...
-          cors: 
-            - enabled: true
-            - origin: '*'
-            - headers:  
-              - Content-Type
-              - Transfer-encoding
+  # TwilioIncommingMessage:
+  #   handler: Personal-Interfaces/Functions/handlerIncommingMessage.handler
+  #   name: TwilioIncommingMessage-lambda-${opt:stage}
+  #   environment:
```
Comment (FlavioAandres) [inline serverless.yml:14]: :machete:

File: shared/database/repos/boxFlow.js [hunk 4]
```diff
@@ -2,18 +2,18 @@ const boxFlowSchema = require("../../models/boxflow");
 
 module.exports.saveBoxFlow = (boxFlow) => {
   return boxFlowSchema.create({
-    phoneNumber: "whatsapp:+573022939843",
+    phoneNumber: "whatsapp:+573008794497",
```
Comment (FlavioAandres) [inline shared/database/repos/boxFlow.js:5]: Hey lets convert this to a environment var :)

File: shared/database/repos/boxFlow.js [hunk 5]
```diff
@@ -28,10 +28,10 @@ module.exports.getByCategories = () => {
 
 module.exports.getByMonth = () => {
   return boxFlowSchema.aggregate([
-    { $match: { phoneNumber: "whatsapp:+573022939843" } },
+    { $match: { phoneNumber: "whatsapp:+573008794497" } },
     {
       $group: {
-        _id: { $substr: ["$createdDate", 0, 7] },
+        _id: { $substr: ["$createdAt", 0, 7] },
```
Comment (FlavioAandres) [inline shared/database/repos/boxFlow.js:34]: Hey lets update the model to remove the createdDate if is not used

**Review by FlavioAandres — CHANGES_REQUESTED** [review]
Little changes requested :)

File: AutomationServices/AutoMailChecker/utils.js [hunk 6]
```diff
@@ -34,30 +34,44 @@ module.exports.readRawEmail = async (emails = []) => {
...
+                    description = value.substring(first, end)
                     textWithValue = description
+                    break;
                 }
+                case "PSE": {
+                    /**
+                     * I want to keep on the description the company name where the payment was made
+                     * This value is before the value of the transaction, but some transaction at the first match value
+                     * has a CUS (like a uuid) of the transaction, that's why I need to look for the value 'Valor de la Transacción'
+                     */
```
Comment (FlavioAandres) [inline AutomationServices/AutoMailChecker/utils.js:62]: Completly agree about this comment! Thanks!

File: config/example.json [hunk 7]
```diff
@@ -6,5 +6,6 @@
     "MONGO_USER": "FlavioAandres",
     "SRV_CONFIG": false,
     "EMAIL_USERNAME": "user@domain.tld",
-    "EMAIL_PASSWORD": "SECRET"
+    "EMAIL_PASSWORD": "SECRET",
+    "PHONE_NUMBER":"+573000000000"
```
Comment (FlavioAandres) [inline config/example.json:10]: We can use this new env var for the furter searchs of boxflow/prepayments 🤘🏼

File: shared/database/repos/boxFlow.js [hunk 8]
```diff
@@ -1,34 +1,37 @@
...
-    phoneNumber: "whatsapp:+573008794497",
+    phoneNumber: `whatsapp:${PHONE_NUMBER}`,
     ...boxFlow,
   });
 };
 
 module.exports.getByCategories = () => {
   return boxFlowSchema.aggregate([
-    { $match: { phoneNumber: "whatsapp:+573008794497" } },
+    { $match: { phoneNumber: `whatsapp:${PHONE_NUMBER}`, type: "EXPENSE" } },
```
Comment (FlavioAandres) [inline shared/database/repos/boxFlow.js:15]: Hey, should I re-populate my data in order to support this 😭😢?

**Review by FlavioAandres — APPROVED** [review]
LGTM 🤘🏼🤘🏼 

Just let me know a summary of all the changes that I should do to maintain my data correctly displayed in frontend and all the schemas in mongo working as it should

File: shared/database/repos/boxFlow.js (hunk 8 above)
Comment (AndresMorelos) [reply shared/database/repos/boxFlow.js:15]: Not Really, you will need to re-deploy your lambdas. Let me upload the last commit.

**Review by FlavioAandres — APPROVED** [review]
LGTM

**Pull request merged**